### PR TITLE
Split ts numbers data classes in different files

### LINKED ISSUE #1333: BC scenarisation : split data for BCs in TSNumberData.cpp
As we have a new source file for BC data definitions, shouldn't we split **src/libs/antares/study/scenario-builder/TSnumberData.cpp** in the same way for load, wind, thermal clusters, etc.

_Originally posted by @guilpier-code in https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1219#discussion_r1210319199_

### PR DESCRIPTION
**Pull request by JasonMarechal25**

### CODE REVIEW

File: src/tests/resources/Antares_Simulator_Tests
Comment (flomnes) [inline src/tests/resources/Antares_Simulator_Tests:1]: Revert ?

File: src/libs/antares/study/scenario-builder/BindingConstraintsTSNumbersData.cpp [hunk 1]
```diff
@@ -30,11 +30,30 @@
...
-#include "TSnumberData.h"
+#include "BindingConstraintsTSNumbersData.h"
+#include "applyToMatrix.hxx"
 
 namespace Antares::Data::ScenarioBuilder
 {
+
+// ================================
+// Binding Constraints ...
+// ================================
```
Comment (guilpier-code) [inline src/libs/antares/study/scenario-builder/BindingConstraintsTSNumbersData.cpp]: We should remove the unnecessary comment / title

**Comment by guilpier-code** [comment]
As @flomnes said, should we revert the changes on sub modules ?

**Review by guilpier-code — DISMISSED** [review]

**Review by sylvlecl — APPROVED** [review]

**Comment by flomnes** [comment]
Yes, please resolve conflicts and revert any changes to submodules

**Review by flomnes — APPROVED** [review]

**Pull request merged**